packages/components/src/components/hds/layout/flex/item.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+import type { AvailableTagNames, AvailableElements } from './types.ts';
+
+export interface HdsLayoutFlexItemSignature {
+  Args: {
+    tag?: AvailableTagNames;
+    basis?: string | 0;
+    grow?: boolean | number | string;
+    shrink?: boolean | number | string;
+    // TODO final name TBD
+    enableCollapseBelowContentSize?: boolean;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: AvailableElements;
+}
+
+export default class HdsLayoutFlexItem extends Component<HdsLayoutFlexItemSignature> {
+  get componentTag(): AvailableTagNames {
+    return this.args.tag ?? 'div';
+  }
+
+  get inlineStyles(): Record<string, unknown> {
+    const inlineStyles: {
+      'flex-basis'?: string;
+      'flex-grow'?: string;
+      'flex-shrink'?: string;
+    } = {};
+
+    // we handle all non-zero cases of `basis` values via inline styles
+    if (typeof this.args.basis === 'string') {
+      inlineStyles['flex-basis'] = this.args.basis;
+    }
+
+    // we handle non-standard cases of `grow` values via inline styles
```

Comment:
@KristinLBradley let me know what you think of this double approach, partially based on CSS class names for the most common use cases, and on inline styles for the non-standard cases

Reply:
Being able to pass "false/true" for the values seems to be helpful & easier to understand/use for consumers so I think that's a worthwhile approach to allow for either.